docs/pages/radiogroup.js

```diff
@@ -0,0 +1,695 @@
+// @flow strict
+import { type Node } from 'react';
+import { RadioGroup } from 'gestalt';
+import GeneratedPropTable from '../components/GeneratedPropTable.js';
+import PageHeader from '../components/PageHeader.js';
+import MainSection from '../components/MainSection.js';
+import Page from '../components/Page.js';
+import { multipledocgen, type DocGen } from '../components/docgen.js';
+
+export default function DocsPage({
+  generatedDocGen,
+}: {|
+  generatedDocGen: {| [string]: DocGen |},
+|}): Node {
+  return (
+    <Page title="RadioGroup">
+      <PageHeader
+        name="RadioGroup"
+        description={generatedDocGen.RadioGroup?.description}
+        defaultCode={`
+      function RadioButtonExample() {
+        const [favorite, setFavorite] = React.useState(undefined);
+
+        return (
+          <RadioGroup legend="Gender">
+            <RadioGroup.RadioButton
+              checked={favorite === 'Female'}
+              id="genderFemale"
+              label="Female"
+              name="gender-pref"
+              onChange={() => setFavorite( 'Female' )}
+              value="Female"
+            />
+            <RadioGroup.RadioButton
+              checked={favorite === 'Male'}
+              id="genderMale"
+              label="Male"
+              name="gender-pref"
+              onChange={() => setFavorite( 'Male' )}
+              value="Male"
+            />
+            <RadioGroup.RadioButton
+              checked={favorite === 'Non-binary'}
+              id="genderNon-binary"
+              label="Non-binary"
+              name="gender-pref"
+              onChange={() => setFavorite( 'Non-binary' )}
+              value="Non-binary"
+            />
+            <RadioGroup.RadioButton
+              checked={favorite === 'Prefer not to state'}
+              id="genderPrefer not to state"
+              label="Prefer not to state"
+              name="gender-pref"
+              onChange={() => setFavorite( 'Prefer not to state' )}
+              value="Prefer not to state"
+            />
+          </RadioGroup>
+        );
+      }
+      `}
+      />
+
+      <GeneratedPropTable generatedDocGen={generatedDocGen.RadioGroup} />
+      <GeneratedPropTable
```

Comment:
Check out our other sub-component docs; wrapping this in a MainSection and a Subsection allows us to display the sub-component's description and improves the formatting a bit:
```js
<MainSection name="Subcomponents">
  <MainSection.Subsection
    title={generatedDocGen?.FlexItem?.displayName}
    description={generatedDocGen?.FlexItem?.description}
  >
    <GeneratedPropTable
      generatedDocGen={generatedDocGen.FlexItem}
      id={generatedDocGen?.FlexItem?.displayName}
      name={generatedDocGen?.FlexItem?.displayName}
    />
  </MainSection.Subsection>
</MainSection>
```

Reply:
![Screen Shot 2022-05-18 at 7 40 48 PM](https://user-images.githubusercontent.com/10593890/169172639-6a47309b-1662-41bb-8b27-c7970fd5382f.png)

All other cmps have the Subcomponent section right before Variants